test/support/protocol_test_case.rb

```diff
@@ -50,6 +50,51 @@ def run_protocol_scenario program, dap: true, cdp: true, &scenario
       end
     end
 
+    def attach_to_dap_server
+      @sock = Socket.unix @remote_info.sock_path
+      @seq = 1
+      @reader_thread = Thread.new do
+        while res = recv_response
+          @queue.push res
+        end
+      rescue Detach
+      end
+      sleep 0.001 while @reader_thread.status != 'sleep'
+      @reader_thread.run
+      INITIALIZE_DAP_MSGS.each{|msg| send(**msg)}
+    end
+
+    def attach_to_cdp_server
+      sock = Socket.tcp HOST, @remote_info.port
+
+      Timeout.timeout(TIMEOUT_SEC) do
+        sleep 0.001 until @remote_info.debuggee_backlog.join.include? 'Connected'
+      end
+
+      @web_sock = WebSocketClient.new sock
+      @web_sock.handshake @remote_info.port, '/'
+      @id = 1
+      @reader_thread = Thread.new do
+        while res = @web_sock.extract_data
+          @queue.push res
+        end
+      rescue Detach
+      end
+      sleep 0.001 while @reader_thread.status != 'sleep'
+      @reader_thread.run
+      INITIALIZE_CDP_MSGS.each{|msg| send(**msg)}
```

Comment:
Please add the following lines here.
https://github.com/ruby/debug/blob/master/test/support/protocol_test_case.rb#L415-L416

Reply:
The reason for adding this here is as follows:

1. call `attach_to_cdp_server` **directory**.
2. call something that uses @crt_frames
3. @crt_frames is not the latest frame because it may be changed every time 'Debugger.paused' is called.

Reply:
I think `Debugger.paused` is only returned at the first connection, not the second?

```
Loaded suite test/protocol/disconnect_cdp_test
Started
start initial attach
{:id=>2, :result=>{}}
{:method=>"Runtime.executionContextCreated", :params=>{:context=>{:id=>"eae712537be02fa944b64faad339a2ab", :origin=>"http://127.0.0.1:63585", :name=>""}}}
{:id=>3, :result=>{}}
{:id=>4, :result=>{}}
{:id=>8, :result=>{}}
{:method=>"Debugger.scriptParsed", :params=>{:scriptId=>"1", :url=>"/private/var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debug-20221106-84811-rdzgps.rb", :startLine=>0, :startColumn=>0, :endLine=>12, :endColumn=>0, :executionContextId=>1, :hash=>"2195685978487654025"}}
{:method=>"Debugger.paused", :params=>{:reason=>"other", :callFrames=>[{:callFrameId=>"0fb3aefabd50b783487268956071f58c", :functionName=>"<main>", :functionLocation=>{:lineNumber=>0, :scriptId=>"1"}, :location=>{:lineNumber=>0, :scriptId=>"1"}, :url=>"/private/var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debug-20221106-84811-rdz
gps.rb", :scopeChain=>[{:type=>"local", :object=>{:type=>"object", :objectId=>"0.449851775437477"}}, {:type=>"script", :object=>{:type=>"object", :objectId=>"0.07042992169684925"}}, {:type=>"global", :object=>{:type=>"object", :objectId=>"0.7769977982489024"}}], :this=>{:type=>"object"}}]}}
end initial attach
disconnected
start second attach
{:id=>2, :result=>{}}
{:method=>"Runtime.executionContextCreated", :params=>{:context=>{:id=>"bfb3534d760e52e0db4d88a84ea031a4", :origin=>"http://127.0.0.1:63585", :name=>""}}}
{:id=>3, :result=>{}}
{:id=>4, :result=>{}}
{:id=>8, :result=>{}}
F
=============================================================================================================================================================================================================================================================================================================================================
Failure: test_closing_cdp_connection_doesnt_kill_the_debuggee(DEBUGGER__::DisconnectCDPTest):
```

That's why I didn't move those 2 lines as it'll break reconnected cases like `test/protocol/disconnect_cdp_test`

Reply:
Oh, seriously? Thank you for checking it. I'll also check it later.

Reply:
I just checked the behavior, and you are right. Also, because I noticed that this behavior is a bug, I need to fix it, too.

Reply:
If it's OK with you, could you write comments somewhere as follows?

```
TODO: Chrome DevTools doesn't support reattaching. We need to fix it.
```

Reply:
Sure. But should I also remove the reattaching part from `test/protocol/disconnect_cdp_test.rb` as it's actually not working?

Reply:
> But should I also remove the reattaching part from test/protocol/disconnect_cdp_test.rb as it's actually not working?

If you can, yes.

Reply:
I found that we still need to reattach to the CDP server in disconnection test because we need to terminate it. But I added the comment right above the attaching call so others can understand that's not how actual reattaching works.

Reply:
Thank you!